src/plugins/intel_npu/src/plugin/npuw/llm_compiled_model.cpp

```diff
@@ -0,0 +1,361 @@
+    }
+    return result;
+}
+} // namespace
+
+ov::npuw::LLMCompiledModel::LLMCompiledModel(const std::shared_ptr<ov::Model>& model,
+                                             const std::shared_ptr<const ov::IPlugin>& plugin,
+                                             const ov::AnyMap& properties)
+    : ov::ICompiledModel(model, plugin),
+     m_options_desc(std::make_shared<::intel_npu::OptionsDesc>()),
+     m_cfg(m_options_desc) {
+    LOG_VERB(__PRETTY_FUNCTION__);
+    LOG_BLOCK();
+
+    ::intel_npu::registerNPUWOptions(*m_options_desc);
+
+    std::map<std::string, ov::Any> npuw_dyn_llm_props;
+    std::map<std::string, ov::Any> other_props;
+    split_llm_properties(properties, npuw_dyn_llm_props, other_props);
+    m_cfg.update(any_copy(npuw_dyn_llm_props));
+
+    // (1) Make template model to be kvcache model, used in generation phase
+    auto kvcache_model = model->clone();
+    // (2) Expose KV-cache input and output layers from kvcache model
+    ov::pass::StatefulToStateless().run_on_model(kvcache_model);
+
+    // (3) Create prefill model from passed template model
+    auto prefill_model = kvcache_model->clone();
+    prefill_model->set_friendly_name(kvcache_model->get_friendly_name() + "_prefill");
+    // (4) Convert prefill model to fp16
+    prefill_model = cvt_kvcache_to_fp16(prefill_model);
+
+    // (5) Optimize kvcache model to output key/values for new token
+    kvcache_model = redirect_new_kv_to_output(kvcache_model);
+    // (6) Convert kvcache model to fp16
+    kvcache_model = cvt_kvcache_to_fp16(kvcache_model);
+
+    const uint32_t kMaxPromptLen = align_to(m_cfg.get<::intel_npu::NPUW_LLM_MAX_PROMPT_LEN>(), 64u);
+    const uint32_t kMinResponseLen = align_to(m_cfg.get<::intel_npu::NPUW_LLM_MIN_RESPONSE_LEN>(), 64u);
+    const ::intel_npu::npuw::dynamic_llm::ModelDesc model_desc = m_cfg.get<::intel_npu::NPUW_LLM_MODEL_DESC>();
+    KVAxesPosition axes = get_kv_axes(model_desc.type);
+    m_kvcache_desc = KVCacheDesc
+        { kMaxPromptLen, kMaxPromptLen + kMinResponseLen, 0u, axes.seq_len};
+    // (7) Make prefill model with static shapes
+    reshape_to_static(prefill_model, m_kvcache_desc.max_prompt_size, m_kvcache_desc.max_prompt_size, axes);
+    // (8) Make kvcache model with static shapes
+    reshape_to_static(kvcache_model, 1u, m_kvcache_desc.total_size, axes);
+
+    auto npudesc = extract_npu_descriptor(plugin);
+
+    ov::AnyMap properties_copy = other_props;
+    auto prefill_config = pop_or_default(
+        properties_copy, "PREFILL_CONFIG", get_default_prefill_config(model, npudesc)
+    // NB: GENERATE_HINT is only applicable for default generate config!
+    );
+    const ::intel_npu::npuw::dynamic_llm::GenerateHint generate_hint = m_cfg.get<::intel_npu::NPUW_LLM_GENERATE_HINT>();
+    auto generate_config = pop_or_default(
+        properties_copy, "GENERATE_CONFIG", get_default_generate_config(model, npudesc, generate_hint)
+    );
```

Comment:
I believe there were comments about this part, too?

It won't work - there's no such properties registered - so please use the configs you have built-in here.

Reply:
Yes, sorry, my fault. I forgot to push changes :(